server/gitrest/packages/gitrest-base/src/utils/helpers.ts

```diff
@@ -254,12 +254,40 @@ export async function checkSoftDeleted(
     const lumberjackProperties = {
         ...getLumberjackBasePropertiesFromRepoManagerParams(repoManagerParams),
     };
+    const metric = Lumberjack.newLumberMetric(
+        GitRestLumberEventName.CheckSoftDeleted,
+        lumberjackProperties);
     const softDeletedMarkerPath = getSoftDeletedMarkerPath(repoPath);
-    const softDeleteBlobExists = await exists(fileSystemManager, softDeletedMarkerPath);
-    const softDeleted = softDeleteBlobExists !== false && softDeleteBlobExists.isFile();
-    if (softDeleted) {
-        const error = new NetworkError(410, "The requested resource has been deleted.");
-        Lumberjack.error("Attempted to retrieve soft-deleted document.", lumberjackProperties, error);
+    try {
+        const softDeleteBlobExists = await exists(fileSystemManager, softDeletedMarkerPath);
+        const softDeleted = softDeleteBlobExists !== false && softDeleteBlobExists.isFile();
+        metric.setProperties({ softDeleted });
+        metric.success("Checked if document is soft-deleted.");
+        if (softDeleted) {
+            const error = new NetworkError(410, "The requested resource has been deleted.");
+            Lumberjack.error("Attempted to retrieve soft-deleted document.", lumberjackProperties, error);
+            throw error;
+        }
+    } catch (e) {
+        metric.error("Failed to check if document is soft-deleted.", e);
```

Comment:
I actually realized that the following situation can happen:
1. We check if the document is soft deleted. The operation succeeds, and then we mark the metic as "success".
2. Then, since the document was soft deleted, we enter the if statement on line 266
3. We throw the error on line 269
4. That would activate the catch block below, and override the metric, marking it as "error".

Would that behavior be ok? Marking it an error after marking it as success would be a no-op, since the metric would have already completed. We would also see a console log line alerting about that: [link](https://github.com/microsoft/FluidFramework/blob/55dcb0efff0b77ab06b87127956266f619d8a679/server/routerlicious/packages/services-telemetry/src/lumber.ts#L133)

I missed this earlier when I made the other comment. One possible solution would be to have the `if (softDeleted) { ... }` check happen outside of the try/catch. Another option would be to add some logic in the catch block to avoid marking the metric as error if the error is a `NetworkError` with status 410, but the first approach seems cleaner - as long as it would not break the existing logic. What do you think?

Reply:
Good points, again! I made the easy change of making softDeleted a `let`, so that it can be defined and read outside of the try..catch. Let me know if that works for you.